### PR TITLE
Check if repo exists before creating pipeline

### PR DESCRIPTION
**Pull request by bitethecode**
Link: https://github.com/woodpecker-ci/woodpecker/issues/1257
I make sure to confirm the repo exists first before triggering the build and wrote a test to cover it.

### CODE REVIEW

File: server/store/datastore/build.go [hunk 1]
```diff
@@ -104,6 +106,17 @@ func (s storage) CreateBuild(build *model.Build, procList ...*model.Proc) error
 		return err
 	}
 
+	// check if repo exists first
+	var repo model.Repo
+
+	if _, err := sess.SQL("SELECT * FROM `repos` WHERE repo_id = ?", build.RepoID).Get(&repo); err != nil {
```
[the remark this answers is not in the document]
Comment (bitethecode) [reply server/store/datastore/build.go:112]: Sounds good, thank for the feedback :)

**Comment by anbraten** [comment]
Nice approach. Be careful with your code and don't forget to merge master as this is a bit behind. (build was renamed to pipeline)

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
@bitethecode i first resolved merge conflicts and then adjust code to:
 - not use raw sql but xorm to check if repo exist
 - create a new error that can be checked if `errors.Is(createPipelineError, types.RecordNotExist)` as true

**Comment by 6543** [comment]
thanks for the init pull :)

**Comment by bitethecode** [comment]
@6543 Oh I was actually working on that merge conflicts and xorm. Okay, if that is done by your, I would stop

**Comment by 6543** [comment]
> @6543 it's kinda useless to approve your own commits :D

hehe - yes was waiting to see your response ;)

**Comment by 6543** [comment]
> @6543 Oh I was actually working on that merge conflicts and xorm. Okay, if that is done by your, I would stop

oh sorry - if you want just adopt from me what you want and forcepushe it ...

**Comment by 6543** [comment]
@bitethecode ☝️

**Comment by bitethecode** [comment]
No issue here! Actually the implementation was almost identical so I'm okay with what you did! 
@6543 btw, could you recommend any good issues I can pick up? I like to contribute more to this project!

**Comment by 6543** [comment]
hmm good question, we should chat in our chat via Discord/Matrix in the "Development" channel/room about it ;)

**Comment by 6543** [comment]
@bitethecode I did miss to update the func names for the test ... (build -> pipeline)

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
```diff
diff --git a/server/store/datastore/pipeline_test.go b/server/store/datastore/pipeline_test.go
index 22db15e5..1a17beb2 100644
--- a/server/store/datastore/pipeline_test.go
+++ b/server/store/datastore/pipeline_test.go
@@ -303,9 +303,12 @@ func TestPipelines(t *testing.T) {
 }
 
 func TestPipelineIncrement(t *testing.T) {
-       store, closer := newTestStore(t, new(model.Pipeline))
+       store, closer := newTestStore(t, new(model.Pipeline), new(model.Repo))
        defer closer()
 
+       assert.NoError(t, store.CreateRepo(&model.Repo{ID: 1, Owner: "1", Name: "1", FullName: "1/1"}))
+       assert.NoError(t, store.CreateRepo(&model.Repo{ID: 2, Owner: "2", Name: "2", FullName: "2/2"}))
+
        pipelineA := &model.Pipeline{RepoID: 1}
        if !assert.NoError(t, store.CreatePipeline(pipelineA)) {
                return
```

**Pull request merged**